<launch>
    <arg name="output" default="screen"/>
    
    <rosparam param="mm_to_pixel">7</rosparam>

    <rosparam param="phantom_low_p">[0,0,143]</rosparam>
    <rosparam param="phantom_high_p">[167,102,255]</rosparam>
    <rosparam param="inserter_low_p">[39,121,29]</rosparam>
    <rosparam param="inserter_high_p">[165, 255, 255]</rosparam>
    <rosparam param="cam_width">1088</rosparam>
    <rosparam param="cam_height">996</rosparam>
    <rosparam param="cam_offset_x">416</rosparam>
    <rosparam param="cam_offset_y">130</rosparam>

    <rosparam param="mm_to_pixel">7</rosparam>
    <include file="$(find pylon_camera)/launch/pylon_camera_node.launch"/>

    <node type="transform_camera_frame.py" name="transform_camera_frame" pkg="adaptive_ctrl" output="$(arg output)"/>
    <node name="image_processsor" pkg="adaptive_ctrl" type="image_processor.py" output="$(arg output)"/>
    <node name="process_costmap" pkg="adaptive_ctrl" type="process_costmap.py" output="$(arg output)"/>
    <include file="$(find adaptive_ctrl)/launch/path_planner.launch"/>
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find adaptive_ctrl)/rviz/test_overlay.rviz"/>

</launch>